openflights.js

```diff
@@ -2234,215 +2246,223 @@ function exportFlights(type, newWindow) {
-      } else {
-        lat += "&deg;N";
-      }
-      if (lon < 0) {
-        lon = -lon + "&deg;W";
-      } else {
-        lon += "&deg;E";
-      }
-      table +=
-        "<tr><td>" +
-        col[0] +
-        '</td><td><a href="#" onclick="JavaScript:selectAirport(' +
-        col[2] +
-        ');">' +
-        col[1] +
-        "</a> (" +
-        lat +
-        " " +
-        lon +
-        ")</td></tr>";
-    }
-    table += "</table>";
-    bigtable += table + '</td><td style="vertical-align: top">';
+  if (statsData.total.segments === 0) {
+    result.append(gt.gettext("This user has no flights."));
+    return;
+  }
 
-    table = '<table style="border-spacing: 10px 0px">';
-    table +=
-      "<tr><th>" +
-      gt.gettext("Class") +
-      "</th><th>" +
-      gt.gettext("Reason") +
-      "</th><th>" +
-      gt.gettext("Seats") +
-      "</th></tr>";
-    table += "<tr>";
-    table += '<td><div id="chart_class"></div></td>';
-    table += '<td><div id="chart_reason"></div></td>';
-    table += '<td><div id="chart_seat"></div></td>';
-    table += "</tr>";
+  const displayUnit = distance_units[statsData.distance_unit];
 
-    table +=
-      "<tr><th>" +
-      gt.gettext("Class by distance") +
-      "</th><th>" +
-      gt.gettext("Mode") +
-      "</th><th><!-- Empty Cell --></th></tr>";
-    table += "<tr>";
-    table += '<td><div id="chart_class_distance"></div></td>';
-    table += '<td><div id="chart_mode"></div></td>';
-    table += "<td><!-- Empty --></td>";
-    table += "</tr>";
-    table += "<tr><td>";
-    table += "</td><td>";
-    table += "</td><td>";
-    // Empty Cell
-    table += "</td></tr>";
-
-    table += "</table>";
-    bigtable += table + "</td></tr></table>";
+  var bigtable =
+    '<table><td style="vertical-align: top"><img src="/img/close.gif" onclick="JavaScript:closePane();" width=17 height=17></td><td style="vertical-align: top">';
+
+  var table = '<table style="border-spacing: 10px 0px">';
+  table += "<tr><th colspan=2>" + gt.gettext("Unique") + "</th></tr>";
+  for (const dataPoint of [
+    ["Airports", "airports"],
+    ["Carriers", "carriers"],
+    ["Countries", "countries"],
+    ["Vehicles", "vehicles"],
+  ]) {
+    const row = document.createElement("tr");
+    const label = document.createElement("td");
+    label.append(gt.gettext(dataPoint[0]));
+    const value = document.createElement("td");
+    value.append(statsData.unique[dataPoint[1]]);
+    row.append(label);
+    row.append(value);
+    table += row.outerHTML;
+  }
+  table += "<tr><td>&nbsp;</td></tr>";
+
+  var distance = statsData.total.distance_mi;
+  table += "<tr><th colspan=2>" + gt.gettext("Distance") + "</th></tr>";
+  table +=
+    "<tr>" +
+    `<td>${gt.gettext("Total flown")}</td>` +
+    `<td>${statsData.total.distance} ${displayUnit}</td>` +
+    "</tr>";
+  table +=
+    "<tr>" +
+    `<td>${gt.gettext("Around the world")}</td>` +
+    `<td>${(distance / EARTH_CIRCUMFERENCE).toFixed(2)}x</td>` +
```

Comment:
Another later TODO is possibly not having these defined different in both PHP and JS.... Possibly have PHP passt the constant value through when needed for calculations...